PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/common/utils/WebSocketSSLProxy.java

```diff
@@ -0,0 +1,155 @@
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.utils;
+
+import com.neovisionaries.ws.client.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+public class WebSocketSSLProxy {
+    private static final Logger logger = Logger.getLogger(WebSocketSSLProxy.class.getName());
+    private static final int DEFAULT_BUFFER_SIZE = 8192;
+    private int bufferSize = DEFAULT_BUFFER_SIZE;
+    private int connectTimeout = 0;
+    private String webSocketServerUri;
+    private String id;
+    private String password;
+    private ServerSocket serverSocket;
+    private Thread thread;
+    private WebSocket webSocket;
+
+    public WebSocketSSLProxy(String webSocketServerUri, String id, String password) {
+        this.webSocketServerUri = webSocketServerUri;
+        this.id = id;
+        this.password = password;
+    }
+
+    public void start() throws IOException {
+        close();
+        // InetAddress.getByName(null) points to the loopback address (127.0.0.1)
+        serverSocket = new ServerSocket(0, 1, InetAddress.getByName(null));
+        thread = new Thread(() -> {
+            try {
+                for (;;) {
+                    Socket clientSocket = serverSocket.accept();
+                    createWebSocketToSocket(clientSocket);
+                    pipeSocketDataToWebSocket(clientSocket);
+                }
+            } catch (IOException | WebSocketException e) {
+                handleConnectBroken(e);
+            }
+
+        });
+        thread.setName("WebsocketSSLProxy-" + thread.getId());
+        thread.start();
+    }
+
+    public void pipeSocketDataToWebSocket(Socket socket) throws IOException {
+        byte[] buffer = new byte[getBufferSize()];
+        while (true) {
+            int bytesRead = socket.getInputStream().read(buffer);
+            if (bytesRead == -1) {
+                break;
+            }
+
+            webSocket.sendBinary(Arrays.copyOfRange(buffer, 0, bytesRead));
+        }
+    }
+
+    public void createWebSocketToSocket(Socket client) throws IOException, WebSocketException {
+        this.webSocket = new WebSocketFactory().setConnectionTimeout(connectTimeout).createSocket(webSocketServerUri)
+                                               .setUserInfo(this.id, this.password)
+                                               .addListener(new WebSocketAdapter() {
+                                                   @Override
+                                                   public void onBinaryMessage(WebSocket websocket, byte[] bytes) {
+                                                       try {
+                                                           client.getOutputStream().write(bytes);
```

Comment:
`pipeWebSocketOutputToSocket` ?

Reply:
because this method initializes the websocket, and we cannot separate the create code apart, so I think it is better to have the `create` in method name.